chainerx_cc/chainerx/native/native_device/activation.cc

```diff
@@ -14,6 +14,9 @@ namespace chainerx {
 namespace native {
 
 void NativeDevice::IfLessElseASSA(const Array& x1, Scalar x2, Scalar pos, const Array& neg, const Array& out) {
+    const Array& x1_cast = x1.dtype() == out.dtype() ? x : x1.AsType(out.dtype());
+    const Array& pos_cast = pos.dtype() == out.dtype() ? x : pos.AsType(out.dtype());
+    const Array& neg_cast = neg.dtype() == out.dtype() ? x : neg.AsType(out.dtype());
```

Comment:
If `chainerx::Maximum` takes integer array and float scalar as parameters, integer array will cast to float twice. Can we avoid this?